api/src/main/java/org/openmrs/module/fhir2/api/util/FhirUtils.java

```diff
@@ -144,28 +145,46 @@ public static Optional<OpenmrsEncounterType> getOpenmrsEncounterType(Encounter e
 	 * @return localization for the given metadata, from message source, in the authenticated locale
 	 */
 	public static String getMetadataTranslation(OpenmrsMetadata metadata) {
-		// This code is from the REST module which derived it from the UI framework
+		
 		String className = metadata.getClass().getSimpleName();
+		String uuid = metadata.getUuid();
 		
-		// in case this is a proxy, strip off anything after an underscore
-		// ie: EncounterType_$$_javassist_26 needs to be converted to EncounterType
-		int underscoreIndex = className.indexOf("_$");
-		if (underscoreIndex > 0) {
-			className = className.substring(0, underscoreIndex);
+		String localization = getLocalization(className, uuid);
+		
+		if (localization == null) {
+			return metadata.getName();
+		} else {
+			return localization;
 		}
+	}
+	
+	public static String getMetadataTranslation(Concept concept) {
```

Comment:
Concept is not OpenmrsMetadata, and has a slightly different format, so I had to extract this out